### DISCUSSION TITLE
[Proposal] New optional file naming convention

### DISCUSSION THREAD

**Discussion by jecsham** [vercel/next.js#37601] [Ideas]
Hi, on this occasion I would like to create a proposal with the intention of improving (even more) the Nextjs development experience.

### Optional file naming convention

![basic-example](https://user-images.githubusercontent.com/10292582/172934024-c83102ba-29af-43b8-b3d6-9570fe213259.png)

The proposal is to **optionally** support a new file naming convention specifically for `layout` and `page` files.

The main rule is that the file must contain the name of its parent directory, separated by `_`, followed by its functional name (`layout` or `page`).

The final name would have the sanake case style.

### Examples:
- app/dashboard/dashboard_layout.tsx
- app/dashboard/dashboard_page.tsx

And because it would be optional, the root layout/page could ignore this new naming convention.
- app/layout.tsx
- app/page.tsx

### ...But why?
As I said before, the intention is simply to improve the development experience.

In medium to large projects it is common to end up with your code editor full of open files with the same name, so sometimes it can get confusing.

### Additional notes
Rust (lang) also faced this "problem"! => [Rust 2018 Path changes | No more mod.rs](https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#no-more-modrs)

**Comment by vindecodex** [comment]
I'm not sure if this will fit in your case but have you try the rewrites which can be configured on `next.config.js`. 

Link with the docs: https://nextjs.org/docs/api-reference/next.config.js/rewrites

**Comment by leerob** [comment]
Let's move this discussion to the already open RFC for layouts 👍 

https://github.com/vercel/next.js/discussions/37136
